mm/kfence/core.c

```diff
@@ -372,12 +372,24 @@ static void for_each_canary(int index, void (*fn)(unsigned long))
 		fn(addr);
 }
 
+/* The static key to set up a KFENCE allocation. */
+DEFINE_STATIC_KEY_FALSE(kfence_allocation_key);
+
+/* Gates the allocation, ensuring only one succeeds in a given period. */
+static atomic_t allocation_gate = ATOMIC_INIT(1);
+/* Wait queue to wake up heartbeat timer task. */
+static DECLARE_WAIT_QUEUE_HEAD(allocation_wait);
+
 void *kfence_guarded_alloc(struct kmem_cache *cache, size_t override_size, gfp_t gfp)
 {
 	unsigned long flags;
 	void *obj = NULL, *ret;
 	struct kfence_freelist *item;
 	int index = -1;
+	/*
+	 * TODO(glider): for allocations made before RNG initialization prandom_u32_max() will
```

Comment:
I think this doesn't need to be a TODO. We can just 'Note: ' this, because if we do not address it for v1, I don't think anybody will care. We can motivate this also by adding "We still benefit from enabling KFENCE as early as possible, even when the RNG is not yet available, as this will allow KFENCE to detect bugs due to earlier allocations. The only downside is that the out-of-bounds accesses detected are deterministic for such allocations."

Reply:
Ok.